hack/update-codegen-dockerized.sh

```diff
@@ -0,0 +1,213 @@
+# update- and verify- scripts.
+${clientgen} --input-base="k8s.io/kubernetes/pkg/apis" --input="${INTERNAL_DIRS_CSV}" "$@"
+${clientgen} --output-base "${KUBE_ROOT}/vendor" --output-package="k8s.io/client-go" --clientset-name="kubernetes" --input-base="k8s.io/kubernetes/vendor/k8s.io/api" --input="${GV_DIRS_CSV}" --go-header-file ${KUBE_ROOT}/hack/boilerplate/boilerplate.generatego.txt "$@"
+
+listergen_internal_apis=(
+$(
+  cd ${KUBE_ROOT}
+  find pkg/apis -maxdepth 2 -name types.go | xargs -n1 dirname | sort
+)
+)
+listergen_internal_apis=(${listergen_internal_apis[@]/#/k8s.io/kubernetes/})
+listergen_internal_apis_csv=$(IFS=,; echo "${listergen_internal_apis[*]}")
+${listergen} --input-dirs "${listergen_internal_apis_csv}" "$@"
+
+listergen_external_apis=(
+$(
+  cd ${KUBE_ROOT}/staging/src
+  find k8s.io/api -name types.go | xargs -n1 dirname | sort
+)
+)
+listergen_external_apis_csv=$(IFS=,; echo "${listergen_external_apis[*]}")
+${listergen} --output-base "${KUBE_ROOT}/vendor" --output-package "k8s.io/client-go/listers" --input-dirs "${listergen_external_apis_csv}" --go-header-file ${KUBE_ROOT}/hack/boilerplate/boilerplate.generatego.txt "$@"
+
+informergen_internal_apis=(
+$(
+  cd ${KUBE_ROOT}
+  find pkg/apis -maxdepth 2 -name types.go | xargs -n1 dirname | sort
+)
+)
+informergen_internal_apis=(${informergen_internal_apis[@]/#/k8s.io/kubernetes/})
+informergen_internal_apis_csv=$(IFS=,; echo "${informergen_internal_apis[*]}")
+${informergen} \
+  --input-dirs "${informergen_internal_apis_csv}" \
+  --internal-clientset-package k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset \
+  --listers-package k8s.io/kubernetes/pkg/client/listers \
+  --go-header-file ${KUBE_ROOT}/hack/boilerplate/boilerplate.generatego.txt \
+  "$@"
+
+informergen_external_apis=(
+$(
+  cd ${KUBE_ROOT}/staging/src
+  # because client-gen doesn't do policy/v1alpha1, we have to skip it too
+  find k8s.io/api -name types.go | xargs -n1 dirname | sort | grep -v pkg.apis.policy.v1alpha1
+)
+)
+
+informergen_external_apis_csv=$(IFS=,; echo "${informergen_external_apis[*]}")
+
+${informergen} \
+  --output-base "${KUBE_ROOT}/vendor" \
+  --output-package "k8s.io/client-go/informers" \
+  --single-directory \
+  --input-dirs "${informergen_external_apis_csv}" \
+  --versioned-clientset-package k8s.io/client-go/kubernetes \
+  --listers-package k8s.io/client-go/listers \
+  --go-header-file ${KUBE_ROOT}/hack/boilerplate/boilerplate.generatego.txt \
+  "$@"
+
+if [[ -z "$(which protoc)" || "$(protoc --version)" != "libprotoc 3."* ]]; then
+  echo "Generating protobuf requires protoc 3.0.0-beta1 or newer. Please download and"
+  echo "install the platform appropriate Protobuf package for your OS: "
+  echo
+  echo "  https://github.com/google/protobuf/releases"
+  echo
+  echo "WARNING: Protobuf changes are not being validated"
+  exit 1
+fi
+
+PROTO_PACKAGES=(
+  k8s.io/api/core/v1
```

Comment:
why this static list? This should be extracted automatically.

Reply:
This particular static list already existed prior to my commit. I had tried removing the static list to see what happens, which generated a lot of protobuf for packages that apparently don't have any generated protobuf committed to the repo. 

I'm not sure if this is valid or not, or why this is the case. I did a little digging through blames and such, but didn't turn up anything certain. To err on the safe side and preserve parity,I broke it up into two pieces from the original whitelist in root/hack/update-protobuf-dockerized.sh, detailed below. 

If you like, I can get examples of the packages that aren't part of this whitelist that would otherwise generate protobuf that is not committed to the repo.

* Entries in the original whitelist that aren't sub-packages that have their own hack scripts which call code-generator/generate-internal-groups.sh remain in this new root/hack/update-protobuf-dockerized.sh

* Entries in the original whitelist that are subpackages that have their own hack scripts which call code-generator/generate-internal-groups.sh are now part of generate-internal-groups.sh